### PR TITLE
Re-add HCB Wrapped 2024 under Flipper flag

### PR DESCRIPTION
**Pull request by YodaLightsabr**
## Summary of the problem
HCB Wrapped no longer exists on HCB

## Describe your changes
Re-adds the following:
- HCB Wrapped banner (under Flipper flag)
- HCB Wrapped React component & page (under Flipper flag)
- HCB Wrapped data download page

### CODE REVIEW

File: app/policies/user_policy.rb [hunk 1]
```diff
@@ -81,4 +81,8 @@ def disable_feature?
     user.admin? || record == user
   end
 
+  def wrapped?
+    user.admin? || record == user
+  end
+
```
Comment (garyhtou) [inline app/policies/user_policy.rb]: is this used anywhere?

File: app/controllers/users/wrapped_controller.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Users
+  class WrappedController < ApplicationController
+    skip_after_action :verify_authorized
+    before_action :set_data
+
+    def index
+      unless Flipper.enabled?(:bank_wrapped, current_user)
```
Comment (garyhtou) [inline app/controllers/users/wrapped_controller.rb]: Before merging, double check that the flipper flag is disabled

File: app/models/metric/hcb/new_events.rb [hunk 3]
```diff
@@ -26,7 +26,7 @@ def calculate
                .not_hidden
                .not_demo_mode
                .approved
-               .where("EXTRACT(YEAR FROM created_at) = ?", 2023)
+               .where("EXTRACT(YEAR FROM events.created_at) = ?", 2023)
```
Comment (garyhtou) [inline app/models/metric/hcb/new_events.rb]: should be switched to 2024 at some point lol

**Review by garyhtou — COMMENTED** [review]
🪄

**Review by garyhtou — COMMENTED** [review]
🪄

File: app/controllers/users/wrapped_controller.rb (hunk 2 above)
Comment (YodaLightsabr) [reply app/controllers/users/wrapped_controller.rb]: Yep

image

File: app/policies/user_policy.rb (hunk 1 above)
Comment (YodaLightsabr) [reply app/policies/user_policy.rb]: Since we switched to the wrapped_controller I don't think so

Comment (YodaLightsabr) [inline app/policies/user_policy.rb]:
```suggestion
```

**Pull request merged**